projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-as-persistent-user.ts

```diff
@@ -160,25 +160,37 @@ export function selectShippingAddress() {
     'GET',
     '/rest/v2/electronics-spa/cms/pages?*/checkout/shipping-address*'
   ).as('getShippingPage');
-  cy.getByText(/proceed to checkout/i).click();
+  cy.get('button.btn-primary')
+    .getByText(/proceed to checkout/i)
+    .click();
   cy.wait('@getShippingPage');
 
   cy.get('.cx-checkout-title').should('contain', 'Shipping Address');
+
+  cy.get('p.cx-checkout-text').should(
+    'contain',
+    'Select your Shipping Address'
+  );
+
   cy.get('cx-order-summary .cx-summary-partials .cx-summary-row')
     .first()
     .find('.cx-summary-amount')
     .should('not.be.empty');
+
   cy.get('.cx-card-title').should('contain', 'Default Shipping Address');
   cy.get('.card-header').should('contain', 'Selected');
 
   cy.route(
     'GET',
     '/rest/v2/electronics-spa/cms/pages?*/checkout/delivery-mode*'
   ).as('getDeliveryPage');
-  cy.get('button.btn-primary').click();
+  cy.get('button.cx-btn.btn-primary')
+    .should('contain', 'Continue')
+    .click();
   cy.wait('@getDeliveryPage')
     .its('status')
     .should('eq', 200);
+  cy.visit(`${Cypress.config('baseUrl')}/checkout/delivery-mode`);
```

Comment:
reason for using visit? 

If this we're doing it this way, we're not testing the flow of checking out

Reply:
Failed to go to next page sometimes when cypress would click on the continue button to move on to the delivery mode page, but was not able to reproduce this manually.

Reply:
If its not reproducible manually, then there must exist a fix for making it work in cypress in terms of flakiness because visiting it doesnt solve the problem

Reply:
There are issues with loading delivery-mode cms. See https://cxjira.sap.com/browse/C3PO-2396